### PR TITLE
DSD-929: SkipNav component

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-929](https://jira.nypl.org/browse/DSD-929)

## This PR does the following:

- This is set for v1.0 so it can be reviewed later, but I wanted to get it in place so @oliviawongnyc can use this in the `Header`.
- Adds the skip navigational links through the `SkipNav` component (maybe should be renamed `SkipNavigation`?).
- The CSS for this was taken from https://webaim.org/techniques/css/invisiblecontent/#skipnavlinks
- For documentation and historical purposes, this was based on the previous iteration of the NYPL Header and the older skip navigation component: https://github.com/NYPL/dgx-skip-navigation-link.

## How has this been tested?

Storybook in its own document page and in the `TemplateAppContainer` storybook page.

## Accessibility concerns or updates

- This is the second part of creating the accessible skip nav links. The first part is in the `TemplateAppContainer` component and is done through the `main`'s element `id` value of `"mainContent"`. The first link in the `SkipNav` component points to this target through the `"#mainContent"` hash.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr963-lnoyreepw1wwbyhqwg4vrf8nccli94ko.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/625f08e72e91387b17f63546

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr963-uij9gqz9e22degoewxpjnbdkanlpkmjh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266aba82e91387b174879c2

File: src/components/AccessibilityGuide/SkipNavigation.stories.mdx [hunk 1]
```diff
@@ -14,19 +14,21 @@ import { getCategory } from "../../utils/componentCategories";
...
-anchor element for the link through the `TemplateAppContainer` component.
-
-The `TemplateAppContainer` component renders as a `main` HTML element with a
-default `id` of `"mainContent"`. This should be used as the anchor element that
-the skip navigation link points to.
+`"#mainContent"` set on the `main` HTML element on the page. When using the
+Reservoir's `SkipNav` and `TemplateAppContainer` components, this accessibility
+combination is automatically handled.
+
+The `SkipNav` component renders two links, the first being the main link pointing
```
Comment (bigfishdesign13) [inline src/components/AccessibilityGuide/SkipNavigation.stories.mdx]: I see that other areas of the `SkipNav` and a11y documentation describe the second link, but I think that should be done here as well.

File: src/components/AccessibilityGuide/SkipNavigation.stories.mdx [hunk 2]
```diff
@@ -14,19 +14,21 @@ import { getCategory } from "../../utils/componentCategories";
 ## General Information
```
Comment (bigfishdesign13) [inline src/components/AccessibilityGuide/SkipNavigation.stories.mdx:14]: Now that we have a `SkipNav` component, do we need this much inofrmation here?  I am concerned about duplicate information here and in other areas of our docs.

File: src/components/SkipNav/SkipNav.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    )}
+  </Story>
+</Canvas>
+
+<ArgsTable story="SkipNav with Controls" />
+
+## Accessibility
+
+To create a useable page for screen readers, a link or a navigational list of
+links should be added before all the content on a page. The Reservoir achieves
```
Comment (bigfishdesign13) [inline src/components/SkipNav/SkipNav.stories.mdx]: Typo…
```The Reservoir Design System achieves…```

File: src/components/Template/Template.stories.mdx [hunk 4]
```diff
@@ -176,11 +181,14 @@ accessibility violation.
...
-be used as the anchor element that the skip navigation link points to. As of
-v0.25.13, the consuming application is responsible for adding the skip
-navigation feature to the application. If your application is using the current
-NYPL Header, it already contains the skip navigation feature but make sure it is
-pointing to the correct anchor element.
+be used as the anchor element that the skip navigation link points to. If your
+application is using the current NYPL Header, it already contains the skip
+navigation feature but make sure it is pointing to the correct anchor element.
+
+If you are using a custom header component or you need to use the Reservoir's
```
Comment (bigfishdesign13) [inline src/components/Template/Template.stories.mdx]: Typo…
```…need to use the Reservoir Design System (DS) `SkipNav`…```

After this is added, the reference to Reservoir under `Template Children Overview` can be updated to just `DS`.

File: src/theme/components/skipNav.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    a: {
+      backgroundColor: "ui.white",
+      height: "1px",
+      left: "-10000px",
+      overflow: "hidden",
+      position: "absolute",
+      top: "auto",
+      width: "1px",
+      // Only display when the user focuses on the links.
+      _focus: {
```
Comment (bigfishdesign13) [inline src/theme/components/skipNav.ts]: Let’s get this a little closer to the styles as it is currently implemented.  Add the following:
```
padding: var(--nypl-space-inset-extranarrow) var(--nypl-space-inset-narrow),
border: 1px solid var(—nypl-colors-ui-gray-dark)
```

File: src/utils/componentCategories.ts [hunk 6]
```diff
@@ -108,7 +108,14 @@ const categories = {
   },
   navigation: {
     title: "Components/Navigation",
-    components: ["Breadcrumbs", "Link", "Menu", "Pagination", "Subnavigation"],
+    components: [
+      "Breadcrumbs",
+      "Link",
+      "Menu",
+      "Pagination",
+      "SkipNav",
```
Comment (bigfishdesign13) [inline src/utils/componentCategories.ts]: I think `SkipNavigation` would be more applicable.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Some questions and some changes.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr963-uij9gqz9e22degoewxpjnbdkanlpkmjh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266aba82e91387b174879c2

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr963-uij9gqz9e22degoewxpjnbdkanlpkmjh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266aba82e91387b174879c2

File: src/components/AccessibilityGuide/SkipNavigation.stories.mdx (hunk 2 above)
Comment (EdwinGuzman) [reply src/components/AccessibilityGuide/SkipNavigation.stories.mdx:14]: Yeah, it seems redundant and I dont see other app-wide a11y guidelines anytime soon. The component a11y section is specific to the component while this specific page is more for how to use it and how it works in an application. That was my original intention. Should I still combine the two pages or keep them separate but refine them?

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr963-uij9gqz9e22degoewxpjnbdkanlpkmjh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266aba82e91387b174879c2

File: src/components/AccessibilityGuide/SkipNavigation.stories.mdx (hunk 2 above)
Comment (bigfishdesign13) [reply src/components/AccessibilityGuide/SkipNavigation.stories.mdx:14]: OK.  After reading your respnse (re: component specific info vs. site-wide standards and patterns), I do think this page is applicable.  And yes, some refinement would be good.

Other potential topics for this section coudl be `image alt text`, `form input labelling` and `color contrast`.

**Comment by EdwinGuzman** [comment]
Updated PR.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr963-uij9gqz9e22degoewxpjnbdkanlpkmjh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266aba82e91387b174879c2

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr963-uij9gqz9e22degoewxpjnbdkanlpkmjh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266aba82e91387b174879c2

**Review by bigfishdesign13 — APPROVED** [review]
UI updates look good.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr963-uij9gqz9e22degoewxpjnbdkanlpkmjh.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6266aba82e91387b174879c2

**Pull request merged**